cpp/arcticdb/storage/lmdb/lmdb_mock_client.cpp

```diff
@@ -0,0 +1,135 @@
+/* Copyright 2024 Man Group Operations Limited
+ *
+ * Use of this software is governed by the Business Source License 1.1 included in the file licenses/BSL.txt.
+ *
+ * As of the Change Date specified in that file, in accordance with the Business Source License, use of this software will be governed by the Apache License, version 2.0.
+ */
+
+#include <arcticdb/storage/lmdb/lmdb_mock_client.hpp>
+
+#include <arcticdb/codec/segment.hpp>
+#include <arcticdb/entity/atom_key.hpp>
+#include <arcticdb/entity/serialized_key.hpp>
+#include <arcticdb/util/string_utils.hpp>
+
+
+namespace arcticdb::storage::lmdb {
+
+std::string MockLmdbClient::get_failure_trigger(
+        const std::string& path,
+        StorageOperation operation_to_fail,
+        int error_code) {
+    return fmt::format("{}#Failure_{}_{}", path, operation_to_string(operation_to_fail), error_code);
+}
+
+std::string lmdb_operation_string(StorageOperation operation) {
```

Comment:
I'd change the return type to either const char* or string_view. No need to allocate a string object because we're dealing with static string literals.

Reply:
Fixed in [2b3b8c3](https://github.com/man-group/ArcticDB/pull/1414/commits/2b3b8c3ab1f2d08b272572f9e3c782918ac2f56b)

Reply:
Your fix for this one still allocates a string doesn't it? As now you hanve

```
raise_key_exists_error(std::string(lmdb_operation_string(...)))
```

Can't we just change raise_key_exists_error to accept a `string_view`?

Reply:
FWIW I don't think we really need to optimize away string allocations for a testing mock. But these are good habits to get in to

Reply:
> FWIW I don't think we really need to optimize away string allocations for a testing mock. But these are good habits to get in to

I agree with both points. In my experience bad habits have the habit of creeping into the main code. I prefer having the same coding standard for everything than to allow some parts to be suboptimal. (Of course, on how important a task is and how much time will it take)